### ISSUE TITLE
Create instances bulk method

### ISSUE THREAD

**Issue by Emilianocm23** [googleapis/python-compute#210]
Hello, it would be nice to have a method to create multiple instances at the same time.
Im not sure if this already exists but i wasnt able to find it

**Comment by busunkim96** [comment]
@m-strzelczyk Is there a nice way to do this with the current API surface?

**Comment by m-strzelczyk** [comment]
I think there are 2 ways you can approach this:

1. Create a [Managed Instance Group (MIG)](https://cloud.google.com/compute/docs/instance-groups) and set its number of replicas to whatever you want. You it's easy to manage and quite flexible way to create/destroy large numbers of VMs.
2. Just use a for-loop to send multiple instance creation requests. You don't have to wait for one request to be completed before sending the next one. Just collect all the operation IDs for your request and after the loop make sure they are all finished.

@Emilianocm23 do you have any specific requirements that wouldn't be met with one of the approaches I have mentioned?

**Comment by Emilianocm23** [comment]
> I think there are 2 ways you can approach this:
> 
> 1. Create a [Managed Instance Group (MIG)](https://cloud.google.com/compute/docs/instance-groups) and set its number of replicas to whatever you want. You it's easy to manage and quite flexible way to create/destroy large numbers of VMs.
> 2. Just use a for-loop to send multiple instance creation requests. You don't have to wait for one request to be completed before sending the next one. Just collect all the operation IDs for your request and after the loop make sure they are all finished.
> 
> @Emilianocm23 do you have any specific requirements that wouldn't be met with one of the approaches I have mentioned?

Thanks for your reply @m-strzelczyk !
I was using the second option, i did a for loop but i need to collect all the instances IPs, so i would need to wait for the response of each instance? or how do i make it faster (please if possible some example code)?

**Comment by Emilianocm23** [comment]
> I wrote this simple code sample, it won't work as is, but should give you a proper idea of what I had in my mind. Yes, in the end you'll need to wait for each instance to be created, but they will be created in parallel, so creating 10 instances should take more or less the same time as creating 1 instance.
> 
> ```python
> import time
> 
> import googleapiclient.discovery
> 
> PROJECT = 'project-name'
> ZONE = 'your-zone'
> 
> def create_instance(compute, project, zone, name, *whateverelse) -> dict:
>     """
>     This is your function creating a new instance, up to you what you
>     put here, just make sure to return the operation ID.
>     """
>     # operation_id = compute.instances().insert(
>     #     project=project,
>     #     zone=zone,
>     #     body=config).execute()
>     # return operation_id
> 
> 
> def wait_for_operation(compute, project, zone, operation):
>     """
>     Wait until the operation is finished or failed.
>     """
>     while True:
>         result = compute.zoneOperations().get(
>             project=project,
>             zone=zone,
>             operation=operation).execute()
> 
>         if result['status'] == 'DONE':
>             if 'error' in result:
>                 raise Exception(result['error'])
>             return result
> 
>         time.sleep(1)
> 
> 
> def create_bulk_instances(number_of_instances):
>     """
>     Create multiple instances in parallel, return their IP addresses.
>     """
>     compute = googleapiclient.discovery.build('compute', 'v1')
>     operations = []
>     instance_names = []
>     for i in range(number_of_instances):
>         operations.append(create_instance(compute, PROJECT, ZONE, f"instance-{i}"))
>         instance_names.append(f"instance-{i}")
> 
>     for operation in operations:
>         wait_for_operation(compute, PROJECT, ZONE, operation['name'])
> 
>     ips = {}
> 
>     for instance_name in instance_names:
>         instance = compute.instances.get(project=PROJECT, zone=ZONE, instance=instance_name).execute()
>         # Internal IP:
>         # ips[instance['name']] = instance['networkInterfaces'][0]['networkIP']
>         # External IP:
>         ips[instance['name']] = instance['networkInterfaces'][0]['accessConfigs'][0]['natIP']
>     
>     return ips
> ```
> 
> Note: The way you acquire the IP address of the instance might be different depending on your setup. I assumed the most default one.

Thank you it works perfectly!
Its a perfect way of doing it
